plugins/com.cubrid.cubridmigration.command/src/com/cubrid/cubridmigration/command/MigrationProgressTracker.java

```diff
@@ -0,0 +1,201 @@
+ * IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ *
+ */
+package com.cubrid.cubridmigration.command;
+
+import com.cubrid.cubridmigration.core.dbobject.Table;
+import com.cubrid.cubridmigration.core.engine.config.MigrationConfiguration;
+import com.cubrid.cubridmigration.core.engine.config.SourceCSVConfig;
+import com.cubrid.cubridmigration.core.engine.config.SourceEntryTableConfig;
+import com.cubrid.cubridmigration.core.engine.config.SourceSQLTableConfig;
+import java.io.File;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class MigrationProgressTracker {
+
+    private final AtomicLong totalRecordUnits = new AtomicLong(0);
+    private final AtomicLong completedRecordUnits = new AtomicLong(0);
+
+    private final Set<String> processingTables = ConcurrentHashMap.newKeySet();
+    private final ConcurrentLinkedQueue<String> tableOrder = new ConcurrentLinkedQueue<>();
+    private final Map<String, Integer> tableIndexMap = new ConcurrentHashMap<>();
+    private final Map<String, TableProgressData> tableProgressMap = new ConcurrentHashMap<>();
+    private final Set<String> changedTables = ConcurrentHashMap.newKeySet();
+    private final AtomicBoolean hasChanges = new AtomicBoolean(false);
+    private final AtomicInteger tableOrderSize = new AtomicInteger(0);
+
+    public void initialize(MigrationConfiguration config) {
+        if (config.sourceIsOnline() || config.sourceIsXMLDump()) {
+            processSourceTables(config.getExpEntryTableCfg(), true, config);
+            processSourceTables(config.getExpSQLCfg(), false, config);
+        } else if (config.sourceIsSQL()) {
+            for (String ss : config.getSqlFiles()) {
+                totalRecordUnits.addAndGet(new File(ss).length());
+            }
+        } else if (config.sourceIsCSV()) {
+            for (SourceCSVConfig scc : config.getCSVConfigs()) {
+                totalRecordUnits.addAndGet(new File(scc.getName()).length());
+            }
+        }
+    }
+
+    private void processSourceTables(
+            Collection<?> tables, boolean isEntryTable, MigrationConfiguration config) {

```

Comment:
NIT: 상속과 제네릭을 이용해 파라미터를 줄이는 것도 좋을 것 같습니다.

```suggestion
    private void processSourceTables(
            Collection<? extends SourceTableConfig> tables, MigrationConfiguration config) {
```

Reply:
확인 후 수정하겠습니다.